basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java

```diff
@@ -1710,10 +1743,12 @@ public static String parseSubject(String subject) {
 			lj.expires = lj.issued + 3600L;
 			lj.deployment_id = getDeploymentId(context_id);
 
-			// TODO: Check through the rolemap logic
-			String lti1_roles = ltiProps.getProperty("roles");
-			if (lti1_roles != null && lti1_roles.contains("Instructor")) {
-				lj.roles.add(LaunchJWT.ROLE_INSTRUCTOR);
+			String lti1_roles = upgradeRoleString(ltiProps.getProperty("roles"));
+			if (lti1_roles != null ) {
+				String[] role_pieces = lti1_roles.split(",");
+				for(String role_piece : role_pieces ) {
+					if ( ! lj.roles.contains(role_piece) ) lj.roles.add(role_piece);
+				}
```

Comment:
Make LaunchJWT.roles a Set, then:
lj.roles.addAll(Arrays.asList(lti1_roles.split(",")));

Reply:
OK - I tend to not like one line magic statements in case there is an NPE in the middle of them :)  But I will fix this.

Reply:
I don't want to change the definition of LaunchJWT to be a Set - the spec says that field is a list and making it a Set feels misleading.  But I re-wrote the code to use a Set rather than a loop+if.  This is a classic case of "A good FORTRAN programmer can write FORTRAN in any language" and "A good perl programmer can write perl in any language" :)

Reply:
I think I have got it in that last commit.

Reply:
good to go? thanks for the Tests